File: Baha/elevation/src/com/esprit/models/Matiere.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.esprit.models;

/**
 *
 * @author NADHIR
 */
public class Matiere {
       private int idMatiere;
    private String nomMatiere;
    private int niveau;

    public Matiere(int idMatiere, String nomMatiere, int niveau) {
        this.idMatiere = idMatiere;
        this.nomMatiere = nomMatiere;
        this.niveau = niveau;
    }

    public Matiere(String nomMatiere) {
        this.nomMatiere = nomMatiere;
    }

    public Matiere(int idMatiere) {
        this.idMatiere = idMatiere;
    }

    public Matiere() {
    }

    public int getIdMatiere() {
        return idMatiere;
    }

    public void setIdMatiere(int idMatiere) {
        this.idMatiere = idMatiere;
    }

    public String getNomMatiere() {
        return nomMatiere;
    }

    public void setNomMatiere(String nomMatiere) {
        this.nomMatiere = nomMatiere;
    }

    public int getNiveau() {
        return niveau;
    }

    public void setNiveau(int niveau) {
        this.niveau = niveau;
    }

    @Override
    public String toString() {
        return "Matiere{" + "idMatiere=" + idMatiere + ", nomMatiere=" + nomMatiere + ", niveau=" + niveau + '}';
    }
    
}

File: src/Entity/Matiere.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Matiere
 *
 * @ORM\Table(name="matiere")
 * @ORM\Entity
 */
class Matiere
{
    /**
     * @var int
     *
     * @ORM\Column(name="idMatiere", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idmatiere;

    /**
     * @var string
     *
     * @ORM\Column(name="nomMatiere", type="string", length=25, nullable=false)
     */
    private $nommatiere;

    /**
     * @var int
     *
     * @ORM\Column(name="niveau", type="integer", nullable=false)
     */
    private $niveau;

    /**
     * @return int
     */
    public function getIdmatiere(): int
    {
        return $this->idmatiere;
    }

    /**
     * @param int $idmatiere
     */
    public function setIdmatiere(int $idmatiere): void
    {
        $this->idmatiere = $idmatiere;
    }

    /**
     * @return string
     */
    public function getNommatiere(): string
    {
        return $this->nommatiere;
    }

    /**
     * @param string $nommatiere
     */
    public function setNommatiere(string $nommatiere): void
    {
        $this->nommatiere = $nommatiere;
    }

    /**
     * @return int
     */
    public function getNiveau(): int
    {
        return $this->niveau;
    }

    /**
     * @param int $niveau
     */
    public function setNiveau(int $niveau): void
    {
        $this->niveau = $niveau;
    }


}
